pandas/_libs/tslibs/offsets.pyx

```diff
@@ -4639,15 +4641,6 @@ cpdef to_offset(freq, bint is_period=False):
                 if not stride:
                     stride = 1
 
-                if prefix in c_DEPR_ABBREVS:
-                    warnings.warn(
-                        f"\'{prefix}\' is deprecated and will be removed in a "
-                        f"future version. Please use \'{c_DEPR_ABBREVS.get(prefix)}\' "
-                        f"instead of \'{prefix}\'.",
-                        FutureWarning,
-                        stacklevel=find_stack_level(),
-                    )
-                    prefix = c_DEPR_ABBREVS[prefix]
```

Comment:
because I wanted to keep this together with line 4668: `offset = _get_offset(prefix) `where we use this prefix. There is no need to do it, I can move it back.

Reply:
yeah that might be better - wouldn't we risk missing the `if prefix in {"D", "H", "min", "s", "ms", "us", "ns"}:` like this?

Reply:
I moved the check` if prefix in c_DEPR_ABBREVS:` back to line 4642 and added a test that fail if the order of checks is wrong.

Reply:
> added a test that fails if the order of the checks isn't correct.

this is music to my ears - well done!